src/ImageSharp/PixelFormats/PixelBlenders/DefaultHardLightPixelBlender{TPixel}.cs

```diff
@@ -0,0 +1,55 @@
+﻿// <copyright file="DefaultHardLightPixelBlender{TPixel}.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp.PixelFormats.PixelBlenders
+{
+    using System;
+    using System.Numerics;
+    using ImageSharp.PixelFormats;
+
+    /// <summary>
+    /// Abstract base class for calling pixel composition functions
+    /// </summary>
+    /// <typeparam name="TPixel">The type of the pixel</typeparam>
+    internal class DefaultHardLightPixelBlender<TPixel> : PixelBlender<TPixel>
+        where TPixel : struct, IPixel<TPixel>
+    {
+        /// <summary>
+        /// Gets the static instance of this blender.
+        /// </summary>
+        public static DefaultHardLightPixelBlender<TPixel> Instance { get; } = new DefaultHardLightPixelBlender<TPixel>();
+
+        /// <inheritdoc />
+        public override TPixel Compose(TPixel background, TPixel source, float amount)
+        {
+            return PorterDuffFunctions<TPixel>.HardLightFunction(background, source, amount);
+        }
+
+        /// <inheritdoc />
+        public override void Compose(BufferSpan<TPixel> destination, BufferSpan<TPixel> background, BufferSpan<TPixel> source, BufferSpan<float> amount)
```

Comment:
It's not that bad. Quite similar to old ASP/ASPX synthax. The only problem is the lack of builtin intellisense support.

Wanna see a real nightmare from the world of metaprogramming? Check out [this](https://github.com/bioinformatics-centre/BayesTyper/blob/master/bayesTyperUtils/vcf%2B%2B/src/boost/variant/variant.hpp)!